include/circt/Dialect/ESI/ESITypes.td

```diff
@@ -0,0 +1,152 @@
+//===- ESITypes.td - ESI data type definitions -------------*- tablegen -*-===//
+//
+// Data types for the ESI dialect
+//
+//===----------------------------------------------------------------------===//
+
+// Base class for other typedefs. Provides dialact-specific defaults
+class ESI_Type<string name> : TypeDef<ESI_Dialect, name> { }
+
+def FixedPoint : ESI_Type<"FixedPoint"> {
+  let summary = "ESI fixed point number type";
+  let description = [{
+    A fixed point number is an approximate representation of a real number --
+    like a floating point number -- with the radix point fixed. While this
+    representation lacks the dynamic range of floating point numbers, it is
+    far simpler and more appropriate for many applications.
+
+    It is typically represented as an `int` with the position of the radix
+    point implictly define in the instructions. ESI supports this
+    representation to not loose this type information. Operations on this
+    type need not be any more complicated than on integer representations.
+  }];
+
+  let mnemonic = "fixed";
+  let parser = [{ AUTOGEN_PARSER($_ctxt, $_parser, FixedPointType, isSigned, wholeBits, fractionalBits); }];
+  let printer = [{ AUTOGEN_PRINTER($_printer, fixed, isSigned, wholeBits, fractionalBits); }];
```

Comment:
I am still concerned by the asmparser/printer logic, but unrelated to that:
nit: plz fit in 80 columns :)